src/Npgsql/Internal/NpgsqlConnector.cs

```diff
@@ -804,25 +817,33 @@ async Task RawOpen(SslMode sslMode, NpgsqlTimeout timeout, bool async, Cancellat
                     var checkCertificateRevocation = Settings.CheckCertificateRevocation;
 
                     RemoteCertificateValidationCallback? certificateValidationCallback;
-                    if (sslMode == SslMode.Prefer || sslMode == SslMode.Require)
+
+                    if (Settings.TrustServerCertificate && sslMode is SslMode.Allow or SslMode.VerifyCA or SslMode.VerifyFull)
```

Comment:
It's impossible to get `SslMode.Allow` here due to `if (sslMode is SslMode.Prefer or SslMode.Require or SslMode.VerifyCA or SslMode.VerifyFull)`.

Reply:
You're right. I'll move this specific check back to NpgsqlConnectionStringBuilder.